packages/esm-patient-forms-app/src/forms/forms-dashboard.component.tsx

```diff
@@ -1,24 +1,61 @@
-import React from 'react';
+import React, { useCallback, useMemo } from 'react';
 import { useConfig, usePatient } from '@openmrs/esm-framework';
-import { useVisitOrOfflineVisit } from '@openmrs/esm-patient-common-lib';
-import type { HtmlFormEntryForm } from '../config-schema';
+import { closeWorkspace, useVisitOrOfflineVisit } from '@openmrs/esm-patient-common-lib';
+import type { ConfigObject } from '../config-schema';
 import FormsList from './forms-list.component';
 import styles from './forms-dashboard.scss';
+import { launchFormEntryOrHtmlForms } from '../form-entry-interop';
+import { useForms } from '../hooks/use-forms';
 
 const FormsDashboard = () => {
-  const config = useConfig();
-  const htmlFormEntryForms: Array<HtmlFormEntryForm> = config.htmlFormEntryForms;
+  const config = useConfig<ConfigObject>();
+  const htmlFormEntryForms = config.htmlFormEntryForms;
   const { patient, patientUuid } = usePatient();
+  const { data: forms, error, mutateForms } = useForms(patientUuid, undefined, undefined, undefined, config.orderBy);
   const { currentVisit } = useVisitOrOfflineVisit(patientUuid);
 
+  const handleFormOpen = useCallback(
+    (formUuid: string, encounterUuid: string, formName: string) => {
+      closeWorkspace('clinical-forms-workspace', true);
+      launchFormEntryOrHtmlForms(
+        currentVisit,
+        formUuid,
+        patient,
+        htmlFormEntryForms,
+        encounterUuid,
+        formName,
+        mutateForms,
+      );
+    },
+    [currentVisit, htmlFormEntryForms, patient, mutateForms],
+  );
+
+  const sections = useMemo(() => {
+    return config.formSections?.map((formSection) => ({
+      ...formSection,
+      availableForms: forms?.filter((formInfo) =>
+        formSection.forms.some((formConfig) => formInfo.form.uuid === formConfig || formInfo.form.name === formConfig),
+      ),
+    }));
+  }, [config.formSections, forms]);
+
   return (
     <div className={styles.container}>
-      <FormsList
-        currentVisit={currentVisit}
-        htmlFormEntryForms={htmlFormEntryForms}
-        patient={patient}
-        patientUuid={patientUuid}
-      />
+      {!Boolean(sections) ? (
```

Comment:
The boolean check doesn't distinguish between arrays that are empty and those that are not. When running a dev server with an empty `formSections` config value, I see an empty page. 

```suggestion
      {sections.length === 0) ? (
```